rust/osm2lanes/src/transform/tags_to_lanes/mod.rs

```diff
@@ -84,17 +90,17 @@ impl std::convert::From<Oneway> for bool {
 #[derive(Copy, Clone, Debug)]
 pub enum Infer<T> {
     None,
+    Guessed(T),
     Default(T),
-    // Locale(T),
-    // Calculated(T),
+    Calculated(T),
     Direct(T),
 }
 
 impl<T> Infer<T> {
     pub fn some(self) -> Option<T> {
         match self {
             Self::None => None,
-            Self::Default(v) | Self::Direct(v) => Some(v),
+            Self::Guessed(v) | Self::Default(v) | Self::Calculated(v) | Self::Direct(v) => Some(v),
```

Comment:
I wonder if there are times when we don't want to use a guessed value (due to it's potentially incorrect non-default nature), but this should be fine for now :)

Reply:
Absolutely. Perhaps a `fn some_confident(self) -> Option<T>`. Ideally guesses taint everything they are used to calculate (with appropriate use of `downgrade` type logic), so that by the time we get to the end, we can decide how to render the result (or when we get to road building), depending on the requirements.